parsers/US_ERCOT.py

```diff
@@ -0,0 +1,379 @@
+
+            if hour_dt not in datapoints_by_date:
+                datapoints_by_date[hour_dt] = {
+                    "storage": StorageMix(),
+                    "production": ProductionMix(),
+                }
+
+            target = "storage" if production_source == "battery" else "production"
+            datapoints_by_date[hour_dt][target].add_value(production_source, hour_value)
+
+    for hour_dt, production_and_storage in datapoints_by_date.items():
+        production = production_and_storage.get("production", ProductionMix())
+        storage = production_and_storage.get("storage", StorageMix())
+
+        production_breakdowns.append(
+            zoneKey=zone_key,
+            datetime=hour_dt,
+            source=SOURCE,
+            production=production,
+            storage=storage,
+        )
+
+    return production_breakdowns
+
+
+def fetch_live_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Session | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    data_json = get_data(url=RT_EXCHANGE_URL, session=session)
+    all_data_points = []
+    sortedZoneKeys = "->".join(sorted([zone_key1, zone_key2]))
+    for item in data_json["data"]:
+        data_point = {}
+        data_point["datetime"] = datetime.fromtimestamp(
+            item["interval"] / 1000
+        ).replace(second=0, tzinfo=TX_TZ)
+        data_point["netFlow"] = sum(item[key] for key in EXCHANGE_MAPPING[zone_key2])
+
+        all_data_points.append(data_point)
+
+    # aggregate data points in the same minute interval
+    aggregated_data_points = []
+    for dt in sorted({item["datetime"] for item in all_data_points}):
+        agg_data_point = {}
+        agg_data_point["datetime"] = dt
+        values_dt = [
+            item["netFlow"] for item in all_data_points if item["datetime"] == dt
+        ]
+        agg_data_point["netFlow"] = sum(values_dt) / len(values_dt)
+        agg_data_point["sortedZoneKeys"] = sortedZoneKeys
+        agg_data_point["source"] = SOURCE
+        aggregated_data_points.append(agg_data_point)
+    validated_data_points = [x for x in all_data_points if validate_exchange(x, logger)]
+    return validated_data_points
+
+
+def fetch_production(
+    zone_key: ZoneKey = ZoneKey("US-TEX-ERCO"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    if target_datetime is None or target_datetime > (
+        datetime.now(tz=timezone.utc) - timedelta(days=1)
+    ).replace(tzinfo=target_datetime.tzinfo if target_datetime else timezone.utc):
+        production = fetch_live_production(
+            zone_key=zone_key, session=session, logger=logger
+        )
+    else:
+        production = fetch_historical_production(
+            zone_key=zone_key,
+            session=session,
+            target_datetime=target_datetime,
+            logger=logger,
+        )
+    return production.to_list()
```

Comment:
Yes. Can I set it to 1 month even though the live production only returns data for today and yesterday? Or does both live and historical have to return the data in that frequency?

Reply:
Yes, that sounds good!
The refetch frequency is only for refetches. They don't have to return data with the same frequency

Reply:
Now the historical production fetches the whole month, if target datatime is Feb then all of the data from Feb will be returned. I set the refetch frequency to 30 days, I hope that is fine even though it is not always exactly 30 days.

Reply:
> Now the historical production fetches the whole month, if target datatime is Feb then all of the data from Feb will be returned. I set the refetch frequency to 30 days, I hope that is fine even though it is not always exactly 30 days.

Maybe it would be safer to set it to 28 days? That way no matter what we should fetch every month, now technically it could skip feb if it's triggered on the "wrong" day.